include/swift/SILOptimizer/Analysis/ArraySemantic.h

```diff
@@ -131,6 +134,10 @@ class ArraySemanticsCall {
   /// Returns true on success, false otherwise.
   bool replaceByValue(SILValue V);
 
+  /// Replace a call to append(contentsOf: ) with a series of
+  /// append(element: ) calls.
+  bool replaceByAppendingValues(SILModule &M, SILFunction* appendFn, llvm::SmallVectorImpl<SILValue> &Vs, ArrayRef<ProtocolConformanceRef> &conformances);
```

Comment:
This method signature seemed really silly to me when I wrote it, and still does. The replacement isn't as straightforward as the one that `replaceByValue(SILValue V)` does. Maybe it doesn't even make sense to have this be a method on `ArraySemantic`.